src/main/java/org/isf/medicalinventory/gui/InventoryWardBrowser.java

```diff
@@ -106,16 +131,57 @@ private void initComponents() {
         panelFooter = getPanelFooter();
         getContentPane().add(panelFooter, BorderLayout.SOUTH);
 
-        for (int i = 0; i < pColumwidth.length; i++) {
-            jTableInventory.getColumnModel().getColumn(i).setMinWidth(pColumwidth[i]);
-        }
+        ajustWidth();
 
         addWindowListener(new WindowAdapter() {
             public void windowClosing(WindowEvent e) {
                 dispose();
             }
         });
-        // pack();
+		pagesCombo.setEditable(true);
+		previous.setEnabled(false);
+		next.setEnabled(false);
+		next.addActionListener(actionEvent -> {
+			if (!previous.isEnabled()) {
+				previous.setEnabled(true);
+			}
+			startIndex += PAGE_SIZE;
+			jTableInventory.setModel(new InventoryBrowsingModel(startIndex, PAGE_SIZE));
+			if ((startIndex + PAGE_SIZE) > totalRows) {
+				next.setEnabled(false);
+			}
+			pagesCombo.setSelectedItem(startIndex / PAGE_SIZE + 1);
+		});
+
+	    previous.addActionListener(actionEvent -> {
+		    if (!next.isEnabled()) {
+			    next.setEnabled(true);
+		    }
+		    startIndex -= PAGE_SIZE;
+		    jTableInventory.setModel(new InventoryBrowsingModel(startIndex, PAGE_SIZE));
+		    if (startIndex < PAGE_SIZE) {
+			    previous.setEnabled(false);
+		    }
+		    pagesCombo.setSelectedItem(startIndex / PAGE_SIZE + 1);
+	    });
+	    pagesCombo.addItemListener(new ItemListener() {
+
+		    public void itemStateChanged(ItemEvent itemEvent) {
+			    int eventID = itemEvent.getStateChange();
+
+			    if (eventID == ItemEvent.SELECTED) {
+				    int page_number = (Integer) pagesCombo.getSelectedItem();
+				    startIndex = (page_number - 1) * PAGE_SIZE;
+
+				    next.setEnabled((startIndex + PAGE_SIZE) <= totalRows);
```

Comment:
```suggestion
				    next.setEnabled(startIndex + PAGE_SIZE <= totalRows);
```
Unnecessary parentheses